src/figuresClasses.ts

```diff
@@ -1,19 +1,112 @@
-export interface Figure {
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Colors = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
 }
 
-export class Triangle {
+abstract class BaseFigure implements Figure {
+  protected isOk: boolean = true;
+
```

Comment:
Is it necessary? You can just throw an error if needed it. Also, I believe that you can find better name )

Reply:
@AlinaMatuschak Good advice) Now it's perfect)